components/public-api-server/pkg/auth/context.go

```diff
@@ -4,22 +4,44 @@
 
 package auth
 
-import "context"
+import (
+	"context"
+	"errors"
+)
 
 type contextKey int
 
 const (
 	authContextKey contextKey = iota
 )
 
-func TokenToContext(ctx context.Context, token string) context.Context {
+type TokenType int
+
+const (
+	AccessTokenType TokenType = iota
+	CookieTokenType
+)
+
+type Token struct {
+	Type  TokenType
+	Value string
+}
+
+func NewAccessToken(token string) Token {
+	return Token{
+		Type:  AccessTokenType,
+		Value: token,
+	}
+}
+
+func TokenToContext(ctx context.Context, token Token) context.Context {
 	return context.WithValue(ctx, authContextKey, token)
 }
 
-func TokenFromContext(ctx context.Context) string {
-	if val, ok := ctx.Value(authContextKey).(string); ok {
-		return val
+func TokenFromContext(ctx context.Context) (Token, error) {
```

Comment:
Because previously we returned the empty string, as we needed to return a primitive - the client doing an empty check seemed reasonable. Now we return a struct, and asking the client to check that is a bit more complex, so we error explicitly to force the client to handle the case.

Reply:
One artifact of my previous choice to return empty string is that tests didn't actually supply a valid token, it was empty. But they were passing